packages/contractkit/src/utils/ber-utils.ts

```diff
@@ -0,0 +1,54 @@
+import * as asn1 from 'asn1js'
+import { BigNumber } from 'bignumber.js'
+import { bigNumberToBuffer, bufferToBigNumber } from './signature-utils'
+
+export const toArrayBuffer = (b: Buffer): ArrayBuffer => {
+  return b.buffer.slice(b.byteOffset, b.byteOffset + b.byteLength)
+}
+
+export function publicKeyFromAsn1(b: Buffer): BigNumber {
+  const { result } = asn1.fromBER(toArrayBuffer(b))
+  const values = (result as asn1.Sequence).valueBlock.value
+  const value = values[1] as asn1.BitString
```

Comment:
just want to make sure it's safe to assume `values.length >= 2`

Reply:
Good point. Either way we'll fail, but it's better to fail with a more descriptive error.